### PR TITLE
functional tests

### PR DESCRIPTION
**Pull request by nicolasaunai**

### CODE REVIEW

File: src/solver/multiphysics_integrator.h [hunk 1]
```diff
@@ -337,7 +338,7 @@ namespace solver
         double getMaxFinerLevelDt(const int /*finerLevelNumber*/, const double coarseDt,
                                   const SAMRAI::hier::IntVector& ratio) override
         {
-            return coarseDt / (ratio.max() * ratio.max()) / 2.;
+            return coarseDt / (ratio.max() * ratio.max()) * 0.4;
```
Comment (Dekken) [inline src/solver/multiphysics_integrator.h:341]: these are not the same, please comment for posterity.

File: src/amr/messengers/hybrid_hybrid_messenger_strategy.h [hunk 2]
```diff
@@ -181,7 +181,6 @@ namespace amr
             magneticInit_.regrid(hierarchy, levelNumber, oldLevel, initDataTime);
             electricInit_.regrid(hierarchy, levelNumber, oldLevel, initDataTime);
             interiorParticles_.regrid(hierarchy, levelNumber, oldLevel, initDataTime);
-            levelGhostParticlesOld_.regrid(hierarchy, levelNumber, oldLevel, initDataTime);
```
Comment (Dekken) [inline src/amr/messengers/hybrid_hybrid_messenger_strategy.h:184]: do we want this still?

File: tests/simulator/test_initialization.py [hunk 3]
```diff
@@ -26,7 +29,7 @@ def getHierarchy(self, interp_order, refinement_boxes, qty, nbr_part_per_cell=10
 
         from pyphare.pharein import global_vars
         global_vars.sim =None
-
+        startMPI()
```
Comment (Dekken) [inline tests/simulator/test_initialization.py:32]: possibly rm as diags are updated for overwriting

File: pyphare/pyphare/pharein/simulation.py [hunk 4]
```diff
@@ -290,31 +290,40 @@ def check_diag_options(**kwargs):
     diag_options = kwargs.get("diag_options", None)
     formats = ["phareh5"]
     if diag_options is not None and "format" in diag_options:
+        mode = diag_options["options"].get("mode", "NOT OVERWIRTE")
```
Comment (Dekken) [inline pyphare/pyphare/pharein/simulation.py:293]: unused variable

File: pyphare/pyphare/simulator/simulator.py [hunk 5]
```diff
@@ -29,6 +35,8 @@ def __init__(self, simulation, auto_dump=True):
         self.cpp_dman = None   # DRAGONS
         self.cpp_dw   = None   # i.e. use weakrefs if you have to ref these.
         self.auto_dump = auto_dump
+        from ._simulator import obj
+        obj = self
```
Comment (Dekken) [inline pyphare/pyphare/simulator/simulator.py:39]: LGTM complains this is unused

maybe 

```
        import pyphare.simulator._simulator as _simulator
        _simulator.obj = self
```

is better

File: src/amr/messengers/hybrid_messenger.h [hunk 6]
```diff
@@ -149,11 +149,11 @@ namespace amr
          * @brief see IMessenger::firstStep
          * @param model
          */
-        void firstNonRootStep(IPhysicalModel& model, SAMRAI::hier::PatchLevel& level,
-                              std::shared_ptr<SAMRAI::hier::PatchHierarchy> const& hierarchy,
-                              double time, double newCoarserTime) final
+        void firstStep(IPhysicalModel& model, SAMRAI::hier::PatchLevel& level,
```
Comment (Dekken) [inline src/amr/messengers/hybrid_messenger.h:152]: possible unnecessary change

File: tests/simulator/test_initialization.py [hunk 7]
```diff
@@ -11,6 +13,7 @@
 import numpy as np
 import unittest
 from ddt import ddt, data, unpack
+import os
```
Comment (Dekken) [inline tests/simulator/test_initialization.py:16]: unused

File: tests/simulator/refinement_boxes.py [hunk 8]
```diff
@@ -9,18 +9,20 @@
...
 
 
 @ddt
 class SimulatorRefineBoxInputs(unittest.TestCase):
-
     def __init__(self, *args, **kwargs):
         super(SimulatorRefineBoxInputs, self).__init__(*args, **kwargs)
         self.simulator = None
+        # so we can delete previous diags only on mpi_rank 0
+        startMPI()
```
Comment (Dekken) [inline tests/simulator/refinement_boxes.py:24]: no longer needed

File: src/amr/messengers/hybrid_messenger.h (hunk 6 above)
Comment (nicolasaunai) [reply src/amr/messengers/hybrid_messenger.h:152]: from the multiphysics integrator standpoint this operation is not necessarily reserved to non root levels, that's (for now) only the case for hybrid stuff but I can't say for other ones...

File: src/amr/messengers/hybrid_hybrid_messenger_strategy.h (hunk 2 above)
Comment (nicolasaunai) [reply src/amr/messengers/hybrid_hybrid_messenger_strategy.h]: wtf... I thought it was back in... :/

Comment (nicolasaunai) [reply src/amr/messengers/hybrid_hybrid_messenger_strategy.h]: rebasing deletes that line silently actually....

File: tests/simulator/refinement_boxes.py (hunk 8 above)
Comment (Dekken) [reply tests/simulator/refinement_boxes.py]: oops https://github.com/nicolasaunai/PHARE/blob/taggingnew/tests/simulator/refinement_boxes.py#L59

**Review by Dekken — APPROVED** [review]
